exporter/sematextexporter/writer.go

```diff
@@ -0,0 +1,255 @@
+}
+
+func newSematextHTTPWriter(logger common.Logger, config *Config, telemetrySettings component.TelemetrySettings) (*sematextHTTPWriter, error) {
+	writeURL, err := composeWriteURL(config)
+	if err != nil {
+		return nil, err
+	}
+	// Detect the hostname
+	hostname, err := os.Hostname()
+	if err != nil {
+		logger.Debug("could not determine hostname, using 'unknown' as os.host")
+		hostname = "unknown"
+	}
+
+	return &sematextHTTPWriter{
+		encoderPool: sync.Pool{
+			New: func() any {
+				e := new(lineprotocol.Encoder)
+				e.SetLax(false)
+				e.SetPrecision(lineprotocol.Nanosecond)
+				return e
+			},
+		},
+		httpClientSettings: config.ClientConfig,
+		telemetrySettings:  telemetrySettings,
+		writeURL:           writeURL,
+		payloadMaxLines:    config.PayloadMaxLines,
+		payloadMaxBytes:    config.PayloadMaxBytes,
+		logger:             logger,
+		hostname:           hostname,
+		token:              config.App_token,
+	}, nil
+}
+
+func composeWriteURL(config *Config) (string, error) {
+	writeURL, err := url.Parse(config.ClientConfig.Endpoint)
+	if err != nil {
+		return "", err
+	}
+	if writeURL.Path == "" || writeURL.Path == "/" {
+		// Assuming a default path for posting metrics
+		writeURL, err = writeURL.Parse("write?db=metrics")
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// // Set any query parameters if needed, depending on the structure of the API
+	queryValues := writeURL.Query()
+
+	writeURL.RawQuery = queryValues.Encode()
+
+	return writeURL.String(), nil
+}
+
+// Start implements component.StartFunc
+func (w *sematextHTTPWriter) Start(ctx context.Context, host component.Host) error {
+	httpClient, err := w.httpClientSettings.ToClient(ctx, host, w.telemetrySettings)
+	if err != nil {
+		return err
+	}
+	w.httpClient = httpClient
+	return nil
+}
+
+func (w *sematextHTTPWriter) NewBatch() otel2influx.InfluxWriterBatch {
+	return newSematextHTTPWriterBatch(w)
+}
+
+var _ otel2influx.InfluxWriterBatch = (*sematextHTTPWriterBatch)(nil)
+
+type sematextHTTPWriterBatch struct {
+	*sematextHTTPWriter
+	encoder      *lineprotocol.Encoder
+	payloadLines int
+}
+
+func newSematextHTTPWriterBatch(w *sematextHTTPWriter) *sematextHTTPWriterBatch {
+	return &sematextHTTPWriterBatch{
+		sematextHTTPWriter: w,
+	}
+}
+
+// EnqueuePoint emits a set of line protocol attributes (metrics, tags, fields, timestamp)
+// to the internal line protocol buffer.
+// If the buffer is full, it will be flushed by calling WriteBatch.
+func (b *sematextHTTPWriterBatch) EnqueuePoint(ctx context.Context, measurement string, tags map[string]string, fields map[string]any, ts time.Time, _ common.InfluxMetricValueType) error {
+	if b.encoder == nil {
+		b.encoder = b.encoderPool.Get().(*lineprotocol.Encoder)
+	}
+	// Add token and os.host tags
+	if tags == nil {
+		tags = make(map[string]string)
+	}
+	tags["token"] = b.token      // Add the Sematext token
+	tags["os.host"] = b.hostname // You can make this dynamic to detect the hostname
```

Comment:
What do you mean with `dynamic` here? Yes we define os.host value here, what this comment says?

Reply:
Sorry
Most of these comments were from when i was debugging
But the hostname is automatically detected